src/features/user/BulkCodes/forms/CreationMethodForm.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { ReactElement, useState, useContext } from 'react';
+import { useRouter } from 'next/router';
+import i18next from '../../../../../i18n';
+import { Button } from '@mui/material';
+
+import BulkCodesForm from './BulkCodesForm';
+import SelectorOption, {
+  SelectorOptionProps,
+} from '../components/SelectorOption';
+import BulkCodesError from '../components/BulkCodesError';
+import { useBulkCode } from '../../../common/Layout/BulkCodeContext';
+import { UserPropsContext } from '../../../common/Layout/UserPropsContext';
+import { BulkCodeMethods } from '../../../../utils/constants/bulkCodeConstants';
+
+const { useTranslation } = i18next;
+
+const CreationMethodForm = (): ReactElement | null => {
+  const router = useRouter();
+  const {
+    bulkMethod,
+    setBulkMethod,
+    setProject,
+    setBulkGiftData,
+    setTotalUnits,
+  } = useBulkCode();
+  const [method, setMethod] = useState<typeof bulkMethod>(bulkMethod);
```

Comment:
It is right but, it can be more declarative by setting it as `BulkCodeMethods | null`

Reply:
done